relay-server/src/actors/processor.rs

```diff
@@ -2315,6 +2316,96 @@ mod tests {
         assert_eq!(new_envelope.items().next().unwrap().ty(), &ItemType::Event);
     }
 
+    #[test]
+    fn test_browser_version_extraction_with_pii_like_data() {
+        let processor = EnvelopeProcessor::new(Arc::new(Default::default()));
+        let event_id = protocol::EventId::new();
+
+        let dsn = "https://e12d836b15bb49d7bbf99e64295d995b:@sentry.io/42"
+            .parse()
+            .unwrap();
+
+        let request_meta = RequestMeta::new(dsn);
+        let mut envelope = Envelope::from_request(Some(event_id), request_meta);
+
+        envelope.add_item({
+            let mut item = Item::new(ItemType::Event);
+            item.set_payload(
+                ContentType::Json,
+                r###"
+                    {
+                        "request": {
+                            "headers": [
+                                ["User-Agent", "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/103.0.0.0 Safari/537.36"]
+                            ]
+                        }
+                    }
+                "###,
+            );
+            item
+        });
+
+        let new_envelope = relay_test::with_system(move || {
+            let mut scrub_config = DataScrubbingConfig::default();
+            // enable all the default scrubbing
+            scrub_config.scrub_data = true;
+            scrub_config.scrub_defaults = true;
+            scrub_config.scrub_ip_addresses = true;
+
+            let mut config = ProjectConfig {
+                datascrubbing_settings: scrub_config,
+                ..Default::default()
+            };
+
+            // Make sure to mask any IP-like looking data
+            let pii_config = PiiConfig::from_json(
+                r##"
+                {
+                    "applications": {
+                        "**": ["@ip:mask"]
+                    }
+                }
+                "##,
+            )
+            .unwrap();
+            config.pii_config = Some(pii_config);
```

Comment:
Similar to previous, can we do something like the following?
```rust
let pii_config = PiiConfig......;
let mut config = ProjectConfig {
    datascrubbing_settings: scrub_config,
    pii_config,
    ..Default::default()
};
```

Reply:
I can move this around